File: backend/event-microservice/Event/Event.Application/Dto/EventDto.cs
﻿using Event.Domain.Entities;

namespace Event.Application.Dto;

public class EventDto
{
    public Guid Id { get; set; }
    public string Name { get; set; } = null!;
    public EventCategory Category { get; set; }
    public DateTimeOffset FromDate { get; set; }
    public string Venue { get; set; } = null!;
    public string? CoverImageUrl { get; set; }
}

File: frontend/lib/data/network/models/event_planr/event/event_dto.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'event_dto.freezed.dart';
part 'event_dto.g.dart';

@freezed
class EventDto with _$EventDto {
  const factory EventDto({
    required String id,
    required String name,
    required int category,
    required DateTime fromDate,
    required String venue,
    required String? coverImageUrl,
  }) = _EventDto;

  factory EventDto.fromJson(Map<String, Object?> json) =>
      _$EventDtoFromJson(json);
}
